experimental/dds/tree2/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -445,26 +445,56 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
 	},
 
 	prune: (change: OptionalChangeset, pruneChild: NodeChangePruner): OptionalChangeset => {
-		if (change.childChanges === undefined) {
-			return change;
-		}
-
 		const prunedChange: OptionalChangeset = {};
 		if (change.fieldChange !== undefined) {
+			const prunedNodeChange =
+				change.fieldChange.newContent?.changes !== undefined
+					? pruneChild(change.fieldChange.newContent.changes)
+					: undefined;
+
+			const prunedFieldChange: OptionalFieldChange = {
+				...change.fieldChange,
+			};
+
+			if (prunedFieldChange.newContent?.changes !== undefined) {
+				prunedFieldChange.newContent = nodeUpdateWithNodeChange(
+					prunedFieldChange.newContent,
+					prunedNodeChange,
+				);
+			}
+
+			if (prunedFieldChange.newContent?.changes === undefined) {
```

Comment:
Isn't that redundant give that `nodeUpdateWithNodeChange` already clears the changes field when it is undefined?